src/config/verification.rs

```diff
@@ -159,7 +164,56 @@ fn build_verification_options_from_flags(
     Ok(Some(verification_config))
 }
 
-pub(crate) async fn build_sigstore_trust_root() -> Result<Option<Arc<SigstoreTrustRoot>>> {
+/// Function that builds the Sigstore trust root used for verification. If a trust-config flag is
+/// provided, it uses that file to load the trust root. Otherwise, it builds the trust root from
+/// Sigstore's TUF repository.
+pub(crate) async fn build_sigstore_trust_root(
+    matches: &ArgMatches,
+) -> Result<Option<Arc<SigstoreTrustRoot>>> {
+    if matches.contains_id("trust-config")
+        && let Some(pki_file) = matches.get_one::<PathBuf>("trust-config")
+    {
+        debug!(
+            "Using user specified Sigstore trust root location: {}",
+            pki_file.display()
+        );
+        if !pki_file.exists() {
+            return Err(anyhow!(
+                "specified Sigstore PKI file does not exist: {}",
+                pki_file.display()
+            ));
+        }
```

Comment:
I think we can skip this check, the error message provided by the next line should be explicit enough